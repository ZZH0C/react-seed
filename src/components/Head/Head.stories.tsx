import { Head } from './Head';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

export default {
  title: 'UI/Head',
  component: Head,
} as ComponentMeta<typeof Head>;

const Template: ComponentStory<typeof Head> = () => <Head />;

export const Primary = Template.bind({});
