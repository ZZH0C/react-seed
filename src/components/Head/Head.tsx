import React from 'react';
import { HeaderLogo } from '../Header_logo/HeaderLogo';
import { HeaderUser } from '../Header_user/HeaderUser';

type HeadProps = React.HTMLAttributes<HTMLElement>;

export const Head: React.FC<HeadProps> = () => {
  return (
    <header className="header">
      <div className="container">
        <div className="row">
          <div className="logoBlock">
            <HeaderLogo />
          </div>
          <HeaderUser />
        </div>
      </div>
    </header>
  );
};
