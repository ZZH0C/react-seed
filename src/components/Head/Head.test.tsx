import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { Head } from './Head';

describe('components/Head', () => {
  const mockProps = {
    children: 'some text',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render and match snapshot', () => {
    expect(shallow(<Head>{mockProps.children}</Head>)).toMatchSnapshot();
  });
});
