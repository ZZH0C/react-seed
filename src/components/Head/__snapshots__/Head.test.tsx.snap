// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Head should render and match snapshot 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      some text
    </div>
  </div>
</header>
`;
